File: clinical-hms-api/app/schemas/audit_log.py
"""
Pydantic schema for audit log read responses.

AuditLogEntry mirrors the AuditLog ORM model and is the response shape for
GET /api/v1/audit-logs/. It is read-only — audit entries are written only by
audit_service.log_action() inside route handlers, never via the API.
"""

from datetime import datetime

from pydantic import BaseModel


class AuditLogEntry(BaseModel):
    id: int
    action: str
    actor_id: int | None
    actor_role: str | None
    entity_type: str | None
    entity_id: int | None
    details: str | None
    timestamp: datetime

    model_config = {"from_attributes": True}

File: clinical-hms-web/src/types/audit_log.ts
/**
 * Audit log domain types — mirrors the backend AuditLogEntry schema.
 *
 * AuditLogEntry is read-only; entries are created server-side by
 * audit_service.log_action() whenever a clinical action occurs.
 * The frontend only ever reads these for display in the admin Audit Log page.
 */

export type AuditLogEntry = {
  id: number
  action: string
  actor_id: number | null
  actor_role: string | null
  entity_type: string | null
  entity_id: number | null
  details: string | null
  timestamp: string
}
